### PR TITLE
Add test for struct default parameter valued

### PR DESCRIPTION
**Pull request by pakrym**
Tests for https://github.com/aspnet/Home/issues/2072

### CODE REVIEW

File: src/Microsoft.Extensions.DependencyInjection.Specification.Tests/breakingchanges.netcore.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+[
```
Comment (pranavkm) [inline src/Microsoft.Extensions.DependencyInjection.Specification.Tests/breakingchanges.netcore.json]: Could we write a different test (just so we leave the existing scenario as is).

**Comment by pakrym** [comment]
Not sure why github does not pickup branch commits, but they there: https://github.com/aspnet/DependencyInjection/commits/pakrym/struct-default-test

File: test/Microsoft.Extensions.DependencyInjection.Tests/Fakes/ClassWithOptionalArgsCtorWithStructs.cs [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+            DateTimeOffset dateTimeOffsetDefault = default(DateTimeOffset),
+            Guid guid = new Guid(),
+            Guid guidDefault = default(Guid),
+            S customStruct = new S(),
+            S customStructDefault = default(S)
+        )
+        {
+        }
+
+        public struct S { }
```
Comment (pranavkm) [inline test/Microsoft.Extensions.DependencyInjection.Tests/Fakes/ClassWithOptionalArgsCtorWithStructs.cs]: `CustomStruct`. `S` is a terrible name.

File: src/Microsoft.Extensions.DependencyInjection.Abstractions/Properties/AssemblyInfo.cs [hunk 3]
```diff
@@ -4,3 +4,4 @@
 using System.Runtime.CompilerServices;
 
 [assembly: InternalsVisibleTo("Microsoft.Extensions.DependencyInjection.Tests, PublicKey=0024000004800000940000000602000000240000525341310004000001000100f33a29044fa9d740c9b3213a93e57c84b472c84e0b8a0e1ae48e67a9f8f6de9d5f7f3d52ac23e48ac51801f1dc950abe901da34d2a9e3baadb141a17c77ef3c565dd5ee5054b91cf63bb3c6ab83f72ab3aafe93d0fc3c2348b764fafb0b1c0733de51459aeab46580384bf9d74c4e28164b7cde247f891ba07891c9d872ad2bb")]
+[assembly: InternalsVisibleTo("Microsoft.Extensions.DependencyInjection, PublicKey=0024000004800000940000000602000000240000525341310004000001000100f33a29044fa9d740c9b3213a93e57c84b472c84e0b8a0e1ae48e67a9f8f6de9d5f7f3d52ac23e48ac51801f1dc950abe901da34d2a9e3baadb141a17c77ef3c565dd5ee5054b91cf63bb3c6ab83f72ab3aafe93d0fc3c2348b764fafb0b1c0733de51459aeab46580384bf9d74c4e28164b7cde247f891ba07891c9d872ad2bb")]
```
Comment (pranavkm) [inline src/Microsoft.Extensions.DependencyInjection.Abstractions/Properties/AssemblyInfo.cs]: What's this for?

Comment (pakrym) [inline src/Microsoft.Extensions.DependencyInjection.Abstractions/Properties/AssemblyInfo.cs]: To get internal `ActivatorUtilities` that come from source package into `DependencyInjection`

Comment (pakrym) [inline src/Microsoft.Extensions.DependencyInjection.Abstractions/Properties/AssemblyInfo.cs]: Having this helper as source package would fix the issue.

Comment (pranavkm) [inline src/Microsoft.Extensions.DependencyInjection.Abstractions/Properties/AssemblyInfo.cs]: Can you add a reference to the package instead? InternalsVisibleTo between source packages is a bit unusual in this codebase for good reasons.

Comment (pakrym) [inline src/Microsoft.Extensions.DependencyInjection.Abstractions/Properties/AssemblyInfo.cs]: Yeah, I'll remove this part

File: src/Microsoft.Extensions.DependencyInjection/ServiceLookup/CallSiteFactory.cs [hunk 4]
```diff
@@ -332,7 +332,7 @@ private IServiceCallSite CreateConstructorCallSite(Type serviceType, Type implem
             {
                 var callSite = CreateCallSite(parameters[index].ParameterType, callSiteChain);
 
-                if (callSite == null && Internal.ActivatorUtilities.TryGetDefaultValue(parameters[index], out var defaultValue))
+                if (callSite == null && Internal.ParameterDefaultValue.TryGetDefaultValue(parameters[index], out var defaultValue))
```
Comment (pranavkm) [inline src/Microsoft.Extensions.DependencyInjection/ServiceLookup/CallSiteFactory.cs]: `using Microsoft.Extensions.Internal`?

**Review by pranavkm — APPROVED** [review]

**Pull request merged**